gcp_variant_transforms/transforms/merge_header_definitions.py

```diff
@@ -14,127 +14,59 @@
-      definition = Definition(
-          val[vcf_header_io.VcfParserHeaderKeyConstants.NUM],
-          val[vcf_header_io.VcfParserHeaderKeyConstants.TYPE])
-      self._infos[key][definition] = [vcf_header.file_name]
-    for key, val in vcf_header.formats.iteritems():
-      definition = Definition(
-          val[vcf_header_io.VcfParserHeaderKeyConstants.NUM],
-          val[vcf_header_io.VcfParserHeaderKeyConstants.TYPE])
-      self._formats[key][definition] = [vcf_header.file_name]
-
-  def __eq__(self, other):
-    return self._infos == other._infos and self._formats == other._formats
-
-  @property
-  def infos(self):
-    return self._infos
-
-  @property
-  def formats(self):
-    return self._formats
-
-
-class _DefinitionsMerger(object):
-  """Class for merging two ``VcfHeaderDefinitions``s."""
-
-  # For the same field definition, save at most `_MAX_NUM_FILE_NAMES` names.
-  _MAX_NUM_FILE_NAMES = 5
-
-  def merge(self, first, second):
-    # type: (VcfHeaderDefinitions, VcfHeaderDefinitions) -> None
-    """Updates ``first``'s definitions with values from ``second``."""
-    if (not isinstance(first, VcfHeaderDefinitions) or
-        not isinstance(first, VcfHeaderDefinitions)):
-      raise NotImplementedError
-    self._merge_definitions(first.infos, second.infos)
-    self._merge_definitions(first.formats, second.formats)
-
-  def _merge_definitions(
-      self,
-      first,  # type: Dict[str, Dict[Definition, List[str]]]
-      second  # type: Dict[str, Dict[Definition, List[str]]]
-      ):
-    # type: (...) -> None
-    """Updates ``first`` by merging values from ``first`` and ``second``."""
-    for key, definitions_to_files_map in second.iteritems():
-      for definition, file_names in definitions_to_files_map.iteritems():
-        first[key].setdefault(definition, [])
-        first[key][definition].extend(str(s) for s in file_names)
-        first[key][definition] = (
-            first[key][definition][:self._MAX_NUM_FILE_NAMES])
+from gcp_variant_transforms.beam_io import vcf_header_io # pylint: disable=unused-import
+
+from gcp_variant_transforms.libs import vcf_header_definitions_merger
 
 
 class _MergeDefinitionsFn(beam.CombineFn):
   """Combiner function for merging definitions."""
 
   def __init__(self, definitions_merger):
-    # type: (_DefinitionsMerger) -> None
+    # type: (vcf_header_definitions_merger.DefinitionsMerger) -> None
     super(_MergeDefinitionsFn, self).__init__()
     self._definitions_merger = definitions_merger
 
   def create_accumulator(self):
-    return VcfHeaderDefinitions()
+    return vcf_header_definitions_merger.VcfHeaderDefinitions()
 
   def add_input(self, source, to_merge):
-    # type: (VcfHeaderDefinitions, VcfHeader) -> VcfHeaderDefinitions
-    return self.merge_accumulators([source,
-                                    VcfHeaderDefinitions(vcf_header=to_merge)])
+    # type: (vcf_header_definitions_merger.VcfHeaderDefinitions,
```

Comment:
This type checking can not be understood by the third party tools. In general, we can add the type in two ways:
- One line type annotation: like https://github.com/googlegenomics/gcp-variant-transforms/blob/master/gcp_variant_transforms/transforms/merge_headers.py#L47
- Multiple lines: example: https://github.com/googlegenomics/gcp-variant-transforms/blob/master/gcp_variant_transforms/transforms/merge_headers.py#L39.

We usually go with the first one, if it can fit into one line. If it cannot, we will adopt the second one. If it is still too long, either define an alias or find a shorter (without sacrificing the readability) name.

Reply:
sooo, I presume that your guys' philosophy is "the least amount of alias as possible", right? I'm complying with it here, but I would personally like to alias VcfHeader as well, for uniformity :/ .

Done.